### PR TITLE
pangea-node-sdk: add more Secure Share docs (GEA-11739)

### PR DESCRIPTION
**Pull request by kenany**

### CODE REVIEW

File: packages/pangea-node-sdk/src/types.ts [hunk 1]
```diff
@@ -2175,42 +2175,112 @@ export namespace Share {
...
+     */
     metadata?: Metadata;
+
+    /**
+     * The ID of a stored object.
+     */
     parent_id?: string;
+
+    /**
+     * An case-sensitive path to an object. Contains a sequence of path segments
```
Comment (pangea-andrest) [inline packages/pangea-node-sdk/src/types.ts:2275]:
```suggestion
     * A case-sensitive path to an object. Contains a sequence of path segments
```

File: packages/pangea-node-sdk/src/types.ts [hunk 2]
```diff
@@ -2350,57 +2612,139 @@ export namespace Share {
...
     storage_pool_id?: string;
+
+    /** Only records where storage_pool_id includes each substring. */
     storage_pool_id__contains?: string[];
+
+    /** Only records where storage_pool_id equals one of the provided substrings. */
     storage_pool_id__in?: string[];
     target?: string;
     target__contains?: string[];
     target__in?: string[];
```
Comment (pangea-andrest) [inline packages/pangea-node-sdk/src/types.ts:2634]:
```suggestion

    /** Only records where target_id equals one of the provided substrings. */
    target__in?: string[];
```

**Review by pangea-andrest — APPROVED** [review]

File: packages/pangea-node-sdk/src/types.ts [hunk 3]
```diff
@@ -2350,57 +2612,139 @@ export namespace Share {
...
+
+    /** Only records where storage_pool_id equals this value. */
     storage_pool_id?: string;
+
+    /** Only records where storage_pool_id includes each substring. */
     storage_pool_id__contains?: string[];
+
+    /** Only records where storage_pool_id equals one of the provided substrings. */
     storage_pool_id__in?: string[];
     target?: string;
```
[the remark this answers is not in the document]
Comment (kenany) [reply packages/pangea-node-sdk/src/types.ts:2632]: Should this be `target_id` then? <https://pangea.cloud/docs/api/share#/v1beta/share/link/list>

File: packages/pangea-node-sdk/src/types.ts [hunk 4]
```diff
@@ -2350,57 +2612,139 @@ export namespace Share {
...
     target__contains?: string[];
```
[the remark this answers is not in the document]
Comment (kenany) [reply packages/pangea-node-sdk/src/types.ts:2633]: Should this be `target_id__contains` then? <https://pangea.cloud/docs/api/share#/v1beta/share/link/list>

File: packages/pangea-node-sdk/src/types.ts (hunk 2 above)
Comment (pangea-andrest) [reply packages/pangea-node-sdk/src/types.ts:2634]: here too

File: packages/pangea-node-sdk/src/types.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (pangea-andrest) [reply packages/pangea-node-sdk/src/types.ts:2632]: Yes

File: packages/pangea-node-sdk/src/types.ts (hunk 1 above)
Comment (kenany) [reply packages/pangea-node-sdk/src/types.ts]: I blame [the docs](https://pangea.cloud/docs/api/share#/v1beta/folder/create)! 😄 
Fixed here now.

File: packages/pangea-node-sdk/src/types.ts (hunk 2 above)
Comment (kenany) [reply packages/pangea-node-sdk/src/types.ts:2640]: Added doc.

File: packages/pangea-node-sdk/src/types.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (kenany) [reply packages/pangea-node-sdk/src/types.ts:2637]: Added doc.

File: packages/pangea-node-sdk/src/types.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (kenany) [reply packages/pangea-node-sdk/src/types.ts:2634]: Added doc.

[the remark this answers is not in the document]
Comment (pangea-andrest) [reply packages/pangea-node-sdk/src/types.ts:2634]: Still missing the `_id` in all of them

**Review by pangea-andrest — CHANGES_REQUESTED** [review]

File: packages/pangea-node-sdk/src/types.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (kenany) [reply packages/pangea-node-sdk/src/types.ts:2634]: I'm just adding docs here, not touching the implementation. But sure, I'll get this done.

File: packages/pangea-node-sdk/src/types.ts [hunk 5]
```diff
@@ -2268,50 +2441,105 @@ export namespace Share {
   }
 
   export interface ListFilter {
+    /**
+     * Only records where the object exists in the supplied parent folder path
+     * name.
+     */
     folder?: string;
     folder__contains?: string[];
     folder__in?: string[];
```
Comment (pangea-andrest) [inline packages/pangea-node-sdk/src/types.ts:2450]: It seems they are not in the docs neither. We should remove `folder__contains` and `folder__in`

**Review by pangea-andrest — APPROVED** [review]

File: packages/pangea-node-sdk/src/types.ts (hunk 5 above)
Comment (kenany) [reply packages/pangea-node-sdk/src/types.ts]: Alright, done.

**Pull request merged**